compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -178,6 +182,13 @@ impl<'context> Elaborator<'context> {
             }
         };
 
+        let generics = self.resolve_struct_turbofish_generics(
+            &struct_type.borrow(),
+            generics,
+            last_segment.generics,
+            Span::from(last_segment.ident.span().end()..last_segment.span.end()),
```

Comment:
Empty span:
```suggestion
            Span::from(last_segment.ident.span().start()..last_segment.span.end()),
```

Reply:
Ah, sorry, this is confusing 😅 

The idea is to have the error point like this:

```
foo::<T>
   ^~~~~
```

Since I used a similar code in another place I'll define a function on `PathSegment` for this, and also document it.

Reply:
The span isn't empty, the intent is to underline only the turbofish and not the variable as well